backend/routes/api/seasons.js

```diff
@@ -23,474 +24,490 @@ import {
+
+        return {
+          dateRangeId: oldDateRange.id,
+          seasonChangeLogId: seasonChangeLog.id,
+          startDateOldValue: oldDateRange.startDate,
+          startDateNewValue: newDateRange.startDate,
+          endDateOldValue: oldDateRange.endDate,
+          endDateNewValue: newDateRange.endDate,
+        };
+      });
 
-    const createChangeLogs = DateChangeLog.bulkCreate(changeLogsToCreate);
+      const createChangeLogs = DateChangeLog.bulkCreate(changeLogsToCreate, {
+        transaction,
+      });
 
-    // Update or create dateRanges
-    const updateDates = DateRange.bulkCreate(dates, {
-      updateOnDuplicate: ["startDate", "endDate", "updatedAt"],
-    });
+      // Update or create dateRanges
+      const updateDates = DateRange.bulkCreate(dateRanges, {
+        updateOnDuplicate: ["startDate", "endDate", "updatedAt"],
 
-    // Delete dateRanges removed by the user
-    const deleteDates = DateRange.destroy({
-      where: {
-        id: {
-          [Op.in]: deletedDateRangeIds,
+        transaction,
+      });
+
+      // Delete dateRanges removed by the user
+      const deleteDates = DateRange.destroy({
+        where: {
+          id: {
+            [Op.in]: deletedDateRangeIds,
+          },
         },
-      },
-    });
 
-    await Promise.all([saveSeason, updateDates, createChangeLogs, deleteDates]);
+        transaction,
+      });
 
-    res.sendStatus(200);
+      await Promise.all([
+        saveSeason,
+        updateDates,
+        createChangeLogs,
+        deleteDates,
+      ]);
+
+      await transaction.commit();
+      res.sendStatus(200);
+    } catch (error) {
+      await transaction.rollback();
+      throw error; // Re-throw to let global error handler catch it
+    }
   }),
 );
 
-// approve
+// Approve
 router.post(
   "/:seasonId/approve/",
   checkPermissions(adminsAndApprovers),
   asyncHandler(async (req, res) => {
-    const { seasonId } = req.params;
-    const { notes, readyToPublish } = req.body;
-
-    const season = await Season.findByPk(seasonId, {
-      include: [
-        {
-          model: FeatureType,
-          as: "featureType",
-          attributes: ["id", "name"],
-        },
-        {
-          model: DateRange,
-          as: "dateRanges",
-          attributes: ["id", "startDate", "endDate", "dateTypeId"],
-          include: [
-            {
-              model: DateType,
-              as: "dateType",
-              attributes: ["id", "name"],
-            },
-          ],
-        },
-      ],
-    });
-
-    if (!season) {
-      const error = new Error("Season not found");
+    const seasonId = Number(req.params.seasonId);
 
-      error.status = 404;
-      throw error;
-    }
+    const transaction = await sequelize.transaction();
 
-    // Create "First come, first served" DateRange if applicable
-    await createFirstComeFirstServedDateRange(season);
+    try {
+      await updateStatus(seasonId, "approved", null, transaction);
 
```

Comment:
Not necessary but it'd be nice if we had status enum values somewhere. Then we can use it like `SEASON_STATUS.APPROVED`. It's because I make typos and forget if the first letter is uppercase/lowercase 😅

```js
const SEASON_STATUS = {
  REQUESTED: "requested",
  APPROVED: "approved",
  PENDING_REVIEW: "pending review",
  // more statuses
};
```